Project Matching and Course Creation

rjaltman opened the pull request:
Resolves #40.

Also includes a fix to let instructors see what courses they run.

tamaraniac committed:
Group matching algorithm created and sort of tested (but only sort of)

tamaraniac committed:
Merge branch 'master' into project-matching

tamaraniac committed:
Added api calls

tamaraniac committed:
starting match groups page

tamaraniac committed:
this almost works

tamaraniac committed:
Added migration for running matching

rjaltman committed:
Fixed itertools import error and listing of courses

tamaraniac committed:
Quick fixes to queries

rjaltman committed:
Added custom sidebars for instructors and organizers

rjaltman committed:
Added space for course create on sidebar of instructor

tamaraniac committed:
Student groups now list

rjaltman committed:
Finished course addition by instructors

rjaltman committed:
Added match groups after merge

rjaltman committed:
Merge branch 'project-matching' of github.com:rjaltman/dogu into project-matching

tamaraniac reviewed (approved):
Looks awesome. I've tested the backend extensively, so I know that works. Thanks for the additions, Robert.

rjaltman committed:
Added projects you were grouped into on homepage, and recommendations header

ddsnowboard committed:
Make migration work on all computers

If your database was in a certain state (a lot of states), the migration
would fail.

ddsnowboard committed:
Merge remote-tracking branch 'refs/remotes/origin/project-matching' into project-matching
